packages/multisig-prover-api/src/msg.rs

```diff
@@ -38,18 +43,86 @@ pub struct InstantiateMsg {
     /// of verifiers before calling UpdateVerifierSet. For example, if this is set to 1, UpdateVerifierSet
     /// will fail unless the registered verifier set and active verifier set differ by more than 1.
     pub verifier_set_diff_threshold: u32,
-    /// Type of encoding to use for signed payload. Blockchains can encode their execution payloads in various ways (ABI, BCS, etc).
-    /// This defines the specific encoding type to use for this prover, which should correspond to the encoding type used by the gateway
-    /// deployed on the destination chain.
-    pub encoder: Encoder,
     /// Public key type verifiers use for signing payload. Different blockchains support different cryptographic signature algorithms (ECDSA, Ed25519, etc).
     /// This defines the specific signature algorithm to use for this prover, which should correspond to the signature algorithm used by the gateway
     /// deployed on the destination chain. The multisig contract supports multiple public keys per verifier (each a different type of key), and this
     /// parameter controls which registered public key to use for signing for each verifier registered to the destination chain.
     pub key_type: KeyType,
-    /// An opaque value created to distinguish distinct chains that the external gateway should be initialized with.
-    /// Value must be a String in hex format without `0x`, e.g. "598ba04d225cec385d1ce3cf3c9a076af803aa5c614bc0e0d176f04ac8d28f55".
-    #[serde(with = "axelar_wasm_std::hex")] // (de)serialization with hex module
-    #[schemars(with = "String")] // necessary attribute in conjunction with #[serde(with ...)]
-    pub domain_separator: Hash,
+    /// Address of a contract responsible for signature verification.
+    /// For detailed information, see [`multisig::msg::ExecuteMsg::StartSigningSession::sig_verifier`]
+    pub sig_verifier_address: Option<String>,
+}
+
+#[cw_serde]
+#[derive(Permissions)]
+pub enum ExecuteMsg {
+    // Start building a proof that includes specified messages
+    // Queries the gateway for actual message contents
+    #[permission(Any)]
+    #[cfg(not(feature = "receive-payload"))]
+    ConstructProof(Vec<CrossChainId>),
+    #[permission(Any)]
+    #[cfg(feature = "receive-payload")]
+    ConstructProof {
+        message_ids: Vec<CrossChainId>,
+        payload_bytes: Vec<HexBinary>,
```

Comment:
add a comment about how the order of payload_bytes needs to match the order of payload hashes associated with the message ids?

Reply:
also, I think full_payloads might be a better name, but I'll leave it up to you.